src/context.ts

```diff
@@ -16,20 +20,20 @@ const contextPages = (
         page.nav = { key: item };
         if (parent) page.nav.parent = parent;
         result.push(page);
+        graph.push({ key: item, page });
       }
     } else {
       const [group, groupItems] = item;
-      for (const page of contextPages(groupItems, dir, pages, group))
-        result.push(page);
+      const subResults = contextPages(groupItems, dir, pages, group);
+      graph.push({ key: group, children: subResults.graph });
+      for (const page of subResults.pages) result.push(page);
     }
   }
-  return result;
+  return { pages: result, graph };
```

Comment:
Probably needs to be cleaned up, there doesn't seem to be a good reason to return both